src/tools/crawling_tools.py

```
@@ -0,0 +1,703 @@
+- get_available_sources: List available crawled sources
+- perform_rag_query: Perform RAG queries on stored content
+- search_code_examples: Search for code examples (agentic RAG)
+"""
+
+import asyncio
+import concurrent.futures
+import json
+import os
+import time
+from typing import Any
+from urllib.parse import urlparse
```

Ruff (0.11.9)

`urllib.parse.urlparse` imported but unused

Remove unused import: `urllib.parse.urlparse`

(F401)